packages/xchain-cosmos/src/util.ts

```diff
@@ -65,31 +74,30 @@ export const getAsset = (denom: string): Asset | null => {
  */
 export const getTxsFromHistory = (txs: TxResponse[], mainAsset: Asset): Tx[] => {
   return txs.reduce((acc, tx) => {
-    let msgs: Msg[] = []
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let msgs: any[]
     if ((tx.tx as RawTxResponse).body === undefined) {
-      msgs = codec.fromJSONString(codec.toJSONString(tx.tx as StdTx)).msg
+      msgs = codec.packCosmosAny(tx.tx).msg
     } else {
-      msgs = codec.fromJSONString(codec.toJSONString((tx.tx as RawTxResponse).body.messages))
+      msgs = codec.packCosmosAny((tx.tx as RawTxResponse).body.messages)
     }
 
     const from: TxFrom[] = []
     const to: TxTo[] = []
     msgs.map((msg) => {
       if (isMsgSend(msg)) {
-        const msgSend = msg as MsgSend
-        const amount = msgSend.amount
-          .map((coin) => baseAmount(coin.amount, 6))
-          .reduce((acc, cur) => baseAmount(acc.amount().plus(cur.amount()), 6), baseAmount(0, 6))
+        const msgSend = msg as proto.cosmos.bank.v1beta1.MsgSend
+        const amount = getCoinAmount(msgSend.amount)
 
         let from_index = -1
 
         from.forEach((value, index) => {
-          if (value.from === msgSend.from_address.toBech32()) from_index = index
+          if (value.from === msgSend.from_address.toString()) from_index = index
```

Comment:
What's the type of `from_address` to have to use `toString()`?

Reply:
fixed